File: candidate-search-backend/Models/Candidate.cs
using System;

namespace CandidateSearch.Api.Models;

public class Candidate
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public string Name { get; set; } = string.Empty;

    public DateOnly DateOfBirth { get; set; }

    public string Gender { get; set; } = string.Empty;

    public string Location { get; set; } = string.Empty;

    public string Education { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; }

    public User User { get; set; } = null!;
}

File: -candidate-search-frontend/src/types/candidate.ts
export interface Candidate {
  id: number;
  name: string;
  age: number;
  gender: string;
  education: string;
  location: string;
  createdAt: string;
  lastLoginAt: string | null;
}

export interface CandidateSearchParams {
  page?: number;
  pageSize?: number;
  minAge?: number | string;
  maxAge?: number | string;
  gender?: string;
  location?: string;
  education?: string;
  createdFrom?: string;
  createdTo?: string;
  lastLoginFrom?: string;
  lastLoginTo?: string;
  sortBy?: string;
  sortOrder?: 'asc' | 'desc' | string;
}

export interface PagedResponse<T> {
  page: number;
  pageSize: number;
  totalCount: number;
  items: T[];
}

export interface ApiResponse<T> {
  success: boolean;
  message: string;
  data: T;
}